parity-crypto/src/hmac/mod.rs

```diff
@@ -78,8 +78,8 @@ enum SignerInner {
 impl<T> Signer<T> {
 	pub fn with(key: &SigKey<T>) -> Signer<T> {
 		match &key.0 {
-			KeyInner::Sha256(k) => Signer(SignerInner::Sha256(Hmac::new(k)), PhantomData),
-			KeyInner::Sha512(k) => Signer(SignerInner::Sha512(Hmac::new(k)), PhantomData),
+			KeyInner::Sha256(key_bytes) => Signer(SignerInner::Sha256(Hmac::<rsha2::Sha256>::new_varkey(key_bytes).unwrap()), PhantomData),
```

Comment:
So `new_varkey` always return `Ok`? Let's use that as a proof in `expect` if that's the case.

Reply:
It does, I think the return type is dictated by the `Mac` trait blanket impl ([here](https://docs.rs/crypto-mac/0.7.0/src/crypto_mac/lib.rs.html#35)), but that's not a great reason tbh.